librz/analysis/var_global.c

```diff
@@ -0,0 +1,370 @@
+// SPDX-FileCopyrightText: 2021 Basstorm <basstorm@nyist.edu.cn>
+// SPDX-License-Identifier: LGPL-3.0-only
+
+#include <rz_analysis.h>
+#include <rz_list.h>
+
+/**
+ * \brief Create a new instance of global variable
+ * 
+ * \param name variable name
+ * \param addr variable address
+ * \param comment variable comment
+ * \return RzAnalysisVarGlobal *
+ */
+RZ_API RZ_OWN RzAnalysisVarGlobal *rz_analysis_var_global_new(RZ_NONNULL const char *name, ut64 addr) {
+	rz_return_val_if_fail(name, NULL);
+	RzAnalysisVarGlobal *glob = RZ_NEW0(RzAnalysisVarGlobal);
+	if (!glob) {
+		return NULL;
+	}
+	glob->name = strdup(name);
+	glob->addr = addr;
+	return glob;
+}
+
+typedef struct {
+	RBNode rb;
+	ut64 addr;
+	ut64 size;
```

Comment:
You can just put a `RzAnalysisVarGlobal *` pointer in here instead of addr+size.
In fact, as another alternative to be even more memory-efficient, you can also put the `RBNode` inside the `RzAnalysisVarGlobal` itself and store them in the tree directly without the need for the additional `GlobalVarNode` struct. See for example `RzAnalysisArchHintRecord`/`arch_hints` as an example.